src/Microsoft.DotNet.Build.Tasks/AddDependenciesToProjectJson.cs

```diff
@@ -0,0 +1,184 @@
+﻿using Microsoft.Build.Utilities;
+using Microsoft.Build.Framework;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    /// <summary>
+    /// Parse a project.json, and add additional dependencies, then write a out new project.json.
+    /// </summary>
+    public class AddDependenciesToProjectJson : Task
+    {
+        private Regex VersionStructureRegex = new Regex(@"(\d+\.\d+\.\d+)-((beta|rc2|rc3)?-?(\d+))");
+        private Regex BuildNumberOverrideStructureRegex = new Regex(@"(beta|rc2|rc3)?-?(\d+)?");
+        private Regex ThreePartVersionRegex = new Regex(@"(\d+\.\d+\.\d+)");
```

Comment:
This project is already importing `NuGet.Versioning` for the validation tasks, so you should be able to use the version parsing classes from there. (Like in [ValidateProjectDependencyVersions.cs#L64](https://github.com/dotnet/buildtools/blob/2b2b0f7/src/Microsoft.DotNet.Build.Tasks/ValidateProjectDependencyVersions.cs#L64)).